### DISCUSSION TITLE
[Feature]: using binaries from `$projectRoot`

### DISCUSSION THREAD

**Discussion by ScottAwesome** [vercel/turborepo#2204] [Ideas]
Basically one downfall that turbo has that NX (seemingly) does not, is that you can't re-use binaries from the project root.

For instance, I must have `eslint`, `vitest`, `vite` etc all installed for every single package in the monorepo, turbo does not resolve the binaries from the project root.

It would be great if we could hoist tooling to the root of the monorepo (so dev dependencies could live there for these purposes) but call them from the projects without resorting to hacks such as the following:

```json
{
  "scripts": {
     "build": "../../node_modules/.bin/vite"
  }
}
```

or duplicating packages for each project.

**Comment by ScottAwesome** [comment]
NX has these kinds of features that allow you to share runs without having to duplicate deps across projects, so I know it can be done.

**Comment by anthonyshew** [comment]
Depending on which package manager you are using and your `.npmrc`/`.yarnrc`, the package manager can choose set up node_modules differently.

Turborepo's goal is to work with package managers rather than edit their functionality so I'm not sure that this is something we would want to do today.
